diff --git a/common/include/net/net_namespace.h b/common/include/net/net_namespace.h
--- a/common/include/net/net_namespace.h
+++ b/common/include/net/net_namespace.h
@@ -58,6 +58,9 @@
 
 struct net_ext {
 	struct net		*net;
+#if IS_ENABLED(CONFIG_IP_VS)
+	struct netns_ipvs	*ipvs;
+#endif
@@ -166,10 +169,6 @@
 #endif
 
 	u64			net_cookie; /* written once */
-
-#if IS_ENABLED(CONFIG_IP_VS)
-	struct netns_ipvs	*ipvs;
-#endif
 #if IS_ENABLED(CONFIG_MPLS)
 	struct netns_mpls	mpls;
@@ -196,6 +195,10 @@
 	};
 } __randomize_layout;
 
+#if IS_ENABLED(CONFIG_IP_VS)
+#define net_ipvs_ptr(_net)	((_net)->ext->ipvs)
+#endif
+
 #include <linux/seq_file_net.h>
diff --git a/common/include/net/ip_vs.h b/common/include/net/ip_vs.h
--- a/common/include/net/ip_vs.h
+++ b/common/include/net/ip_vs.h
@@ -37,7 +37,7 @@
 /* Generic access of ipvs struct */
 static inline struct netns_ipvs *net_ipvs(struct net* net)
 {
-	return net->ipvs;
+	return net_ipvs_ptr(net);
 }
 
 /* Connections' size value needed by ip_vs_ctl.c */
diff --git a/common/net/netfilter/ipvs/ip_vs_core.c b/common/net/netfilter/ipvs/ip_vs_core.c
--- a/common/net/netfilter/ipvs/ip_vs_core.c
+++ b/common/net/netfilter/ipvs/ip_vs_core.c
@@ -2300,7 +2300,7 @@
 	/* Counters used for creating unique names */
 	ipvs->gen = atomic_read(&ipvs_netns_cnt);
 	atomic_inc(&ipvs_netns_cnt);
-	net->ipvs = ipvs;
+	net_ipvs_ptr(net) = ipvs;
 	if (ip_vs_estimator_net_init(ipvs) < 0)
 		goto estimator_fail;
 
@@ -2336,7 +2336,7 @@
 control_fail:
 	ip_vs_estimator_net_cleanup(ipvs);
 estimator_fail:
-	net->ipvs = NULL;
+	net_ipvs_ptr(net) = NULL;
 	return -ENOMEM;
 }
 
@@ -2354,7 +2354,7 @@
 		ip_vs_control_net_cleanup(ipvs);
 		ip_vs_estimator_net_cleanup(ipvs);
 		IP_VS_DBG(2, "ipvs netns %d released\n", ipvs->gen);
-		net->ipvs = NULL;
+		net_ipvs_ptr(net) = NULL;
 	}
 }
