diff --git a/common/include/net/net_namespace.h b/common/include/net/net_namespace.h
--- a/common/include/net/net_namespace.h
+++ b/common/include/net/net_namespace.h
@@ -56,9 +56,13 @@ struct bpf_prog;
 
 /* ABI Hack: 外挂结构体，用于存放 NFTables 以防止 struct net 膨胀 */
 struct net_ext {
+	struct net		*net;
 #if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
 	struct netns_nftables	nft;
 #endif
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+	struct netns_sctp	sctp;
+#endif
 };
 
 
@@ -136,9 +140,6 @@ struct net {
 #if IS_ENABLED(CONFIG_IEEE802154_6LOWPAN)
  	struct netns_ieee802154_lowpan	ieee802154_lowpan;
 #endif
-#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
-	struct netns_sctp	sctp;
-#endif
 #ifdef CONFIG_NETFILTER
  	struct netns_nf		nf;
 #if defined(CONFIG_NF_CONNTRACK) || defined(CONFIG_NF_CONNTRACK_MODULE)
@@ -462,4 +463,8 @@ static inline void net_ns_init(void) {}
 #define net_nft(_net)		((_net)->ext->nft)
 #endif
 
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+#define net_sctp(_net)		((_net)->ext->sctp)
+#endif
+
 #endif /* __NET_NET_NAMESPACE_H */
diff --git a/common/net/core/net_namespace.c b/common/net/core/net_namespace.c
--- a/common/net/core/net_namespace.c
+++ b/common/net/core/net_namespace.c
@@ -46,12 +46,13 @@ static struct net *net_alloc(void)
 	if (!net)
 		goto out_free;
 
-#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
+#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE) || defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
 	net->ext = kzalloc(sizeof(*net->ext), GFP_KERNEL);
 	if (!net->ext) {
 		kmem_cache_free(net_cachep, net);
 		goto out_free;
 	}
+	net->ext->net = net;
 #endif
 
 #ifdef CONFIG_KEYS
@@ -438,7 +439,7 @@ static void net_free(struct net *net)
 	if (net->ucounts)
 		dec_ucount(net->ucounts, UCOUNT_NET_NAMESPACES);
 	kfree(net->key_domain);
-#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
+#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE) || defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
 	kfree(net->ext);
 #endif
 	kmem_cache_free(net_cachep, net);
