diff --git a/common/include/net/net_namespace.h b/common/include/net/net_namespace.h
--- a/common/include/net/net_namespace.h
+++ b/common/include/net/net_namespace.h
@@ -134,7 +134,4 @@ struct net {
 #if IS_ENABLED(CONFIG_IEEE802154_6LOWPAN)
  	struct netns_ieee802154_lowpan	ieee802154_lowpan;
 #endif
-#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
-	struct netns_sctp	sctp;
-#endif
 #ifdef CONFIG_NETFILTER
@@ -188,3 +185,11 @@ struct net {
 	/* Force padding: */
-	struct sock		*:64;
+	/* ABI Hack: 存放独立的 SCTP 结构体防止位移 */
+	union {
+		struct sock		*:64;
+		struct {
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+			struct sctp_ext *sctp_ptr;
+#endif
+		};
+	};
 	struct sock		*:64;
@@ -451,3 +456,10 @@ static inline void net_ns_init(void) {}
 #define net_nft(_net)		((_net)->ext->nft)
 #endif
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+struct sctp_ext {
+	struct net *net;
+	struct netns_sctp sctp;
+};
+#define net_sctp(_net) ((_net)->sctp_ptr->sctp)
+#endif
 #endif /* __NET_NET_NAMESPACE_H */
diff --git a/common/net/core/net_namespace.c b/common/net/core/net_namespace.c
--- a/common/net/core/net_namespace.c
+++ b/common/net/core/net_namespace.c
@@ -419,2 +419,10 @@ static struct net *net_alloc(void)
 	if (!net)
 		goto out_free;
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+	/* 为 SCTP 结构体申请独立内存 */
+	net->sctp_ptr = kzalloc(sizeof(struct sctp_ext), GFP_KERNEL);
+	if (!net->sctp_ptr) {
+		goto out_free_net;
+	}
+	net->sctp_ptr->net = net;
+#endif
@@ -443,3 +451,8 @@ static struct net *net_alloc(void)
 #ifdef CONFIG_KEYS
+out_free_ext:
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+	kfree(net->sctp_ptr);
+out_free_net:
+#endif
 out_free_2:
 	kmem_cache_free(net_cachep, net);
@@ -475,3 +488,6 @@ void net_drop_ns(void *p)
 	if (net && refcount_dec_and_test(&net->passive)) {
 		kfree(rcu_access_pointer(net->gen));
+#if defined(CONFIG_IP_SCTP) || defined(CONFIG_IP_SCTP_MODULE)
+		kfree(net->sctp_ptr);
+#endif
 		/* There should not be any trackers left there. */
